app/services/report_service.rb

```diff
@@ -8,7 +8,7 @@ def call(source_account, target_account, options = {})
     @target_account = target_account
     @status_ids     = options.delete(:status_ids).presence || []
     @comment        = options.delete(:comment).presence || ''
-    @category       = options.delete(:category).presence || 'other'
+    @category       = 'violation' if options[:rule_ids].present? || options.delete(:category).presence || 'other'
```

Comment:
This line will only set `@category`, and only to `violation`, when either rule IDs are present, or category is present, or when string `other`. That is not your intent.

Reply:
```ruby
options[:rule_ids].present? ? 'violation' : options.delete(:category).presence || 'other'
```

Would this work?

Reply:
You need paranthesis around the last part

Reply:
Seems to not be necessary but I added them anyway just now

Reply:
Actually adding parentheses *breaks* the code. Adding rule_ids[] makes the category "other" when parenthesized